optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/common/accessor/gizmo/GizmoAnnotationWrappers.java

```
@@ -0,0 +1,690 @@
+                                    break;
+                                case BOOLEAN:
+                                    annotationValues.put(value.name(),
+                                            value.asBooleanArray());
+                                    break;
+                                case CLASS:
+                                    Class[] valueArr = Arrays.stream(value.asClassArray())
+                                            .map(v -> {
+                                                try {
+                                                    return Class.forName(v.toString());
+                                                } catch (ClassNotFoundException e) {
+                                                    throw new IllegalStateException(e);
+                                                }
+                                            })
+                                            .collect(Collectors.toList()).toArray(new Class[0]);
+                                    annotationValues.put(value.name(),
+                                            valueArr);
+                                    break;
+                                case STRING:
+                                    annotationValues.put(value.name(),
+                                            value.asStringArray());
+                                    break;
+                                case ENUM:
+                                    annotationValues.put(value.name(),
+                                            value.asEnumArray());
+                                    break;
+                                case ARRAY:
+                                case NESTED:
+                                case UNKNOWN:
+                                    break;
+                            }
+                            break;
+                        case NESTED:
+                        case UNKNOWN:
+                            break;
```

All switch statements must have a default that fails fast. For example:

 ```
           switch (type) {
                case EQUAL:
                ...
                default:
                    throw new UnsupportedOperationException("Unsupported joiner type (" + type + ").");
            }
```